qiskit/circuit/controlledgate.py

```diff
@@ -16,26 +16,68 @@
 Controlled unitary gate.
 """
 
+from typing import List, Optional, Union
 from qiskit.circuit.exceptions import CircuitError
 from .gate import Gate
 
 
 class ControlledGate(Gate):
-    """Controlled unitary gate."""
+    """Controlled unitary gate.
+    
+    Attributes:
+        num_ctrl_qubits: The number of control qubits to add.
+        base_gate: An instance of the target unitary to control.
+    """
 
-    def __init__(self, name, num_qubits, params, label=None, num_ctrl_qubits=1,
-                 definition=None):
-        """Create a new gate.
+    def __init__(self, name: str, num_qubits: int, params: List,
+                 label: Optional[str] = None, num_ctrl_qubits: Optional[int] = 1,
+                 definition: Optional[List] = None):
+        """Create a new ControlledGate. In the new gate the first ``num_ctrl_qubits``
+        of the gate are the controls.
 
         Args:
-            name (str): The Qobj name of the gate.
-            num_qubits (int): The number of qubits the gate acts on.
-            params (list): A list of parameters.
-            label (str or None): An optional label for the gate [Default: None]
-            num_ctrl_qubits (int): Number of control qubits.
-            definition (list): list of gate rules for implementing this gate.
+            name: The name of the gate.
+            num_qubits: The number of qubits the gate acts on.
+            params: A list of parameters for the gate.
+            label: An optional label for the gate.
+            num_ctrl_qubits: Number of control qubits.
+            definition: A list of gate rules for implementing this gate. The
+                elements of the list are tuples of (:meth:`~qiskit.circuit.Gate`, [qubit_list],
+                [clbit_list]).
```

Comment:
How about the type hint for `definition` be `Optional[List[Tuple[Gate, List[int], List[int]]]] = None`?

@lcapelluto any ideas?

Reply:
I agree with Soolu, the type should be clear from the type hint. The param documentation can make clarifying comments if necessary

Reply:
Sounds good. I didn't realize one could nest them as much.